### PR TITLE
Updated getIndexInfo() to include Columnstore indexes by using custom query.

### PR DESCRIPTION
**Pull request by Ananya2**
Replaced the use of the sp_statistics stored procedure with a custom query to retrieve index information as the sp_statistics procedure did not return Columnstore indexes, so a query using sys.indexes was implemented as a workaround. 
This new query ensures that all index types (Clustered, NonClustered, Columnstore) are included in the result set. 
Github Issue: https://github.com/microsoft/mssql-jdbc/issues/2546

### CODE REVIEW

File: src/test/java/com/microsoft/sqlserver/jdbc/DatabaseMetadataTest.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                    hasColumnstoreIndex = true;
+                } else if (indexType.equals("CLUSTERED")) {
+                    hasClusteredIndex = true;
+                } else if (indexType.equals("NONCLUSTERED")) {
+                    hasNonClusteredIndex = true;
+                }
+            }
+
+            assertTrue(hasClusteredIndex, "CLUSTERED index found.");
+            assertTrue(hasNonClusteredIndex, "NONCLUSTERED index found.");
```
Comment (machavan) [inline src/test/java/com/microsoft/sqlserver/jdbc/DatabaseMetadataTest.java:99]: the messages  would be "non-clustered index not found" as the assert should raise them when hasClusteredIndex is false

File: src/test/java/com/microsoft/sqlserver/jdbc/DatabaseMetadataTest.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            assertTrue(hasColumnstoreIndex, "COLUMNSTORE index found.");
```
Comment (machavan) [inline src/test/java/com/microsoft/sqlserver/jdbc/DatabaseMetadataTest.java:100]: the messages  would be "columnstore index not found" as the assert should raise them when hasClusteredIndex is false

File: src/test/java/com/microsoft/sqlserver/jdbc/DatabaseMetadataTest.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                    "    sys.schemas sch ON t.schema_id = sch.schema_id " +
+                   
+                    "WHERE t.name = '" + tableName + "' " +
+                          "AND sch.name = '" + schema + "' " +
+                    "ORDER BY " +
+                    "    t.name, i.name, ic.key_ordinal;";
+            rs = stmt.executeQuery(query);
+            System.out.println("Testing query " + rs + " " + rs.next());
+            while (rs.next()) {
+                String indexType = rs.getString("IndexType");
```
Comment (machavan) [inline src/test/java/com/microsoft/sqlserver/jdbc/DatabaseMetadataTest.java:85]: Will be best to validate every single field coming out of getIndexInfo call, with the result of this query.

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java [hunk 4]
```diff
@@ -1197,39 +1197,31 @@ private ResultSet executeSPFkeys(String[] procParams) throws SQLException {
...
+                "i.type_desc AS IndexType, " +
+                "i.is_unique AS IsUnique, " +
+                "c.name AS ColumnName, " +
+                "ic.key_ordinal AS ColumnOrder " +
+                "FROM sys.indexes i " +
+                "INNER JOIN sys.index_columns ic ON i.object_id = ic.object_id AND i.index_id = ic.index_id " +
+                "INNER JOIN sys.columns c ON ic.object_id = c.object_id AND ic.column_id = c.column_id " +
+                "INNER JOIN sys.tables t ON i.object_id = t.object_id " +
+                "INNER JOIN sys.schemas sch ON t.schema_id = sch.schema_id " +
+                "WHERE t.name = '" + table + "' " +
```
Comment (machavan) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java:1221]: Let us check for case sensitivity of table name and schema name. Compare this with existing behavior on what happens with sp_statistics

File: src/test/java/com/microsoft/sqlserver/jdbc/DatabaseMetadataTest.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    public static void setupTests() throws Exception {
+        setConnection();
+    }
+
+    @Test
+    public void testGetIndexInfo() throws SQLException, SQLServerException {
+        ResultSet rs = null;
+        try (Connection connection = getConnection(); Statement stmt = connection.createStatement()) {
+        	TestUtils.dropTableIfExists(tableName, stmt);
+        	String createTableSQL = "CREATE TABLE " + tableName + " (" +
```
Comment (machavan) [inline src/test/java/com/microsoft/sqlserver/jdbc/DatabaseMetadataTest.java:33]: Let us use below approach with init and terminate methods for table creation and doppping as it provides idempotency across repeated test runs

        @BeforeEach
        public void init() throws Exception {
            try (Connection con = getConnection()) {
                con.setAutoCommit(false);
                try (Statement stmt = con.createStatement()) {
                    TestUtils.dropTableIfExists(tableName, stmt);
                    stmt.executeUpdate(
                            "CREATE TABLE " + tableName + " (col1 INT, col2 VARCHAR(" + TEST_STRING.length() + "))");
                }
                con.commit();
            }
        }



       @AfterEach
        public void terminate() throws Exception {
            try (Connection con = getConnection(); Statement stmt = con.createStatement()) {
                try {
                    TestUtils.dropTableIfExists(tableName, stmt);
                } catch (SQLException e) {
                    fail(TestResource.getResource("R_unexpectedException") + e.getMessage());
                }
            }
        }

**Comment by Ananya2** [comment]
Changes have already been requested as a part of https://github.com/microsoft/mssql-jdbc/pull/2566, hence closing the PR.

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java (hunk 4 above)
Comment (Ananya2) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java:1221]: I’ve validated that the previous getIndexInfo was also not case-sensitive, similar to this change, and I have added a test case to validate the scenario. I am closing this PR and have added the same changes in [PR #2566](https://github.com/microsoft/mssql-jdbc/pull/2566). Please review.

**Pull request closed without merging**